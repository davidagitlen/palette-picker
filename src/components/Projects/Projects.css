.Projects {
  font-family: 'Source Code Pro', monospace;
  font-size: 28px;
  height: max-content;
  width: 100%;
}