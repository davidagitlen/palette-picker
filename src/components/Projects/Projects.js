import React from "react";
import "./Projects.css";
import Project from '../Project/Project';

const Projects = ({projects, palettes, editProject, editPalette, trashProject, trashPalette, selectPalette}) => {
  const projectList = projects.map(project => (
    <Project
      key={project.id}
      project={project}
      palettes={palettes}
      editProject={editProject}
      editPalette={editPalette}
      trashProject={trashProject}
      trashPalette={trashPalette}
      selectPalette={selectPalette}
    />
  )); 
  return (
    <section className="Projects">
      {projectList}
    </section>
  )
};

export default Projects;