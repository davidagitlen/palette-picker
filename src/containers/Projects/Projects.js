import React from "react";
import "./Projects.css";
import Project from '../Project/Project';

const Projects = () => {
  return (
    <section className="Projects">
      <h2>Projects</h2>
      <Project />
    </section>
  )
};

export default Projects;
