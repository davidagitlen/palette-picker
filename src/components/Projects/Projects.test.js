import React from 'react';
import { shallow } from 'enzyme';
import Projects from './Projects';

it('initial test to pass CI suite', () => {
  expect(true).toEqual(true);
});

describe('Projects', () => {

  it('should match the snapshot with data passed through correctly', () => {
    const projects =  [
      { id: 1, project: 'Project_One' },
      { id: 2, project: 'Project_Two' }
    ];
    const palettes = [
        {
          id: 1,
          project_name: 'Project_One',
          hex_1: '#565656',
          hex_2: '#787878',
          hex_3: '#909090',
          hex_4: '#A1A1A1',
          hex_5: '#B2B2B2',
          project_id: 1
        },
        {
          id: 2,
          project_name: 'Project_Two',
          hex_1: '#565656',
          hex_2: '#787878',
          hex_3: '#909090',
          hex_4: '#A1A1A1',
          hex_5: '#B2B2B2',
          project_id: 2
        }
      ];
      const wrapper = shallow(
        <Projects 
          key={1}
          projects={projects}
          palettes={palettes}
          editProject={jest.fn()}
          editPalette={jest.fn()}
          trashProject={jest.fn()}
          trashPalette={jest.fn()}
          selectPalette={jest.fn()}
        /> 
      );

    expect(wrapper).toMatchSnapshot();
  });
});