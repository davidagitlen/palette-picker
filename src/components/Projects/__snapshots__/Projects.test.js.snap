// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match the snapshot with data passed through correctly 1`] = `
<section
  className="Projects"
>
  <Project
    editPalette={[MockFunction]}
    editProject={[MockFunction]}
    key="1"
    palettes={
      Array [
        Object {
          "hex_1": "#565656",
          "hex_2": "#787878",
          "hex_3": "#909090",
          "hex_4": "#A1A1A1",
          "hex_5": "#B2B2B2",
          "id": 1,
          "project_id": 1,
          "project_name": "Project_One",
        },
        Object {
          "hex_1": "#565656",
          "hex_2": "#787878",
          "hex_3": "#909090",
          "hex_4": "#A1A1A1",
          "hex_5": "#B2B2B2",
          "id": 2,
          "project_id": 2,
          "project_name": "Project_Two",
        },
      ]
    }
    project={
      Object {
        "id": 1,
        "project": "Project_One",
      }
    }
    selectPalette={[MockFunction]}
    trashPalette={[MockFunction]}
    trashProject={[MockFunction]}
  />
  <Project
    editPalette={[MockFunction]}
    editProject={[MockFunction]}
    key="2"
    palettes={
      Array [
        Object {
          "hex_1": "#565656",
          "hex_2": "#787878",
          "hex_3": "#909090",
          "hex_4": "#A1A1A1",
          "hex_5": "#B2B2B2",
          "id": 1,
          "project_id": 1,
          "project_name": "Project_One",
        },
        Object {
          "hex_1": "#565656",
          "hex_2": "#787878",
          "hex_3": "#909090",
          "hex_4": "#A1A1A1",
          "hex_5": "#B2B2B2",
          "id": 2,
          "project_id": 2,
          "project_name": "Project_Two",
        },
      ]
    }
    project={
      Object {
        "id": 2,
        "project": "Project_Two",
      }
    }
    selectPalette={[MockFunction]}
    trashPalette={[MockFunction]}
    trashProject={[MockFunction]}
  />
</section>
`;
